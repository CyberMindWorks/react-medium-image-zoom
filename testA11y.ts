import axe from 'axe-core'

const testA11y = (html: axe.ElementContext): Promise<boolean> =>
  new Promise((resolve, reject) => {
    axe.run(html, {}, (err, { violations }) => {
      if (err) {
        reject(err)
      } else if (violations.length > 0) {
        reject(violations)
      } else {
        resolve(true)
      }
    })
  })

export default testA11y
