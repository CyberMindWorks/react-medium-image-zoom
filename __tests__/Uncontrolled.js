import React from 'react'
import { act, fireEvent, render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import testA11y from '../testA11y'
import Zoom from '../source'

test('is accessible with defaults & <img />', async () => {
  const { getByLabelText } = render(
    <Zoom>
      <img alt="foo" src="foo.jpg" width="500" />
    </Zoom>
  )
  const openTrigger = getByLabelText('Zoom image')

  expect(openTrigger).toBeVisible()
  expect(await testA11y(document.body)).toEqual(true)

  fireEvent.click(openTrigger)

  const closeTrigger = getByLabelText('Unzoom image')

  expect(closeTrigger).toBeVisible()
  expect(await testA11y(document.body)).toEqual(true)
})

test('is accessible with custom open/close text & <img />', async () => {
  const { container, getByLabelText } = render(
    <Zoom closeText="Close me" openText="Open me">
      <img alt="foo" src="foo.jpg" width="500" />
    </Zoom>
  )
  const openTrigger = getByLabelText('Open me')

  expect(openTrigger).toBeVisible()
  expect(await testA11y(container)).toEqual(true)

  fireEvent.click(openTrigger)

  const closeTrigger = getByLabelText('Close me')

  expect(closeTrigger).toBeVisible()
  expect(await testA11y(document.body)).toEqual(true)
})

test('zooms/unzooms with defaults & <img />', () => {
  jest.useFakeTimers()

  const { getByLabelText, getByRole } = render(
    <Zoom>
      <img alt="foo" src="foo.jpg" width="500" />
    </Zoom>
  )
  const openTrigger = getByLabelText('Zoom image')
  expect(openTrigger).toBeVisible()

  fireEvent.click(openTrigger)

  const closeTrigger = getByLabelText('Unzoom image')
  const modal = getByRole('dialog')

  expect(closeTrigger).toBeVisible()

  fireEvent.click(closeTrigger)

  act(() => {
    jest.runAllTimers()
  })

  expect(closeTrigger).not.toBeInTheDocument()
  expect(modal).not.toBeInTheDocument()
})

test('zooms/unzooms with custom open/close text & <img />', async () => {
  jest.useFakeTimers()

  const { getByLabelText, getByRole } = render(
    <Zoom closeText="Close me" openText="Open me">
      <img alt="foo" src="foo.jpg" width="500" />
    </Zoom>
  )
  const openTrigger = getByLabelText('Open me')
  expect(openTrigger).toBeVisible()

  fireEvent.click(openTrigger)

  const closeTrigger = getByLabelText('Close me')
  const modal = getByRole('dialog')

  expect(closeTrigger).toBeVisible()
  expect(modal).toBeVisible()

  fireEvent.click(closeTrigger)

  act(() => {
    jest.advanceTimersByTime(300)
  })

  expect(closeTrigger).not.toBeInTheDocument()
  expect(modal).not.toBeInTheDocument()
})
