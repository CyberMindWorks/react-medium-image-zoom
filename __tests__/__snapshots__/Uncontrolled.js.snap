// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`is accessible with custom open/close text & <img /> 1`] = `
<body>
  <div>
    <main>
      <div
        class="wrapHidden"
      >
        <img
          alt="foo"
          src="foo.jpg"
          width="500"
        />
        <button
          aria-label="Open me"
          class="trigger btn"
        />
      </div>
    </main>
  </div>
  <div
    aria-modal="true"
    class="overlay"
    role="dialog"
    style="background-color: rgba(255, 255, 255, 0.95); transition-duration: 300ms; z-index: 2147483647;"
  >
    <div
      class="content"
      style="height: 0px; left: 0px; top: 0px; transform: scale(Infinity) translate(0px, 0px); transition-duration: 300ms; width: 0px;"
    >
      <img
        alt="foo"
        src="foo.jpg"
        width="500"
      />
    </div>
    <button
      aria-label="Close me"
      class="trigger btn"
    />
  </div>
</body>
`;

exports[`is accessible with defaults & <img /> 1`] = `
<body>
  <div>
    <main>
      <div
        class="wrap"
      >
        <img
          alt="foo"
          src="foo.jpg"
          width="500"
        />
        <button
          aria-label="Zoom image"
          class="trigger btn"
        />
      </div>
    </main>
  </div>
</body>
`;

exports[`is accessible with defaults & <img /> 2`] = `
<body>
  <div>
    <main>
      <div
        class="wrapHidden"
      >
        <img
          alt="foo"
          src="foo.jpg"
          width="500"
        />
        <button
          aria-label="Zoom image"
          class="trigger btn"
        />
      </div>
    </main>
  </div>
  <div
    aria-modal="true"
    class="overlay"
    role="dialog"
    style="background-color: rgba(255, 255, 255, 0.95); transition-duration: 300ms; z-index: 2147483647;"
  >
    <div
      class="content"
      style="height: 0px; left: 0px; top: 0px; transform: scale(Infinity) translate(0px, 0px); transition-duration: 300ms; width: 0px;"
    >
      <img
        alt="foo"
        src="foo.jpg"
        width="500"
      />
    </div>
    <button
      aria-label="Unzoom image"
      class="trigger btn"
    />
  </div>
</body>
`;

exports[`passes on original transform style 1`] = `
<body>
  <div>
    <div
      class="wrap"
      style="transform: rotate(45deg);"
    >
      <img
        alt="foo"
        src="foo.jpg"
        width="500"
      />
      <button
        aria-label="Zoom image"
        class="trigger btn"
      />
    </div>
  </div>
</body>
`;

exports[`passes on original transform style 2`] = `
<body>
  <div>
    <div
      class="wrapHidden"
      style="transform: rotate(45deg);"
    >
      <img
        alt="foo"
        src="foo.jpg"
        width="500"
      />
      <button
        aria-label="Zoom image"
        class="trigger btn"
      />
    </div>
  </div>
  <div
    aria-modal="true"
    class="overlay"
    role="dialog"
    style="background-color: rgba(255, 255, 255, 0.95); transition-duration: 300ms; z-index: 2147483647;"
  >
    <div
      class="content"
      style="height: 0px; left: 0px; top: 0px; transform: scale(Infinity) translate(0px, 0px) rotate(45deg); transition-duration: 300ms; width: 0px;"
    >
      <img
        alt="foo"
        src="foo.jpg"
        width="500"
      />
    </div>
    <button
      aria-label="Unzoom image"
      class="trigger btn"
    />
  </div>
</body>
`;

exports[`renders without browser environment 1`] = `
<body>
  <div
    class="wrap"
  >
    <img
      alt="foo"
      src="foo.jpg"
      width="500"
    />
    <button
      aria-label="Zoom image"
      class="trigger btn"
    />
  </div>
</body>
`;

exports[`unzooms on scroll 1`] = `
<body>
  <div>
    <div
      class="wrapHidden"
    >
      <img
        alt="foo"
        src="foo.jpg"
        width="500"
      />
      <button
        aria-label="Zoom image"
        class="trigger btn"
      />
    </div>
  </div>
  <div
    aria-modal="true"
    class="overlay"
    role="dialog"
    style="background-color: rgba(255, 255, 255, 0.95); transition-duration: 300ms; z-index: 2147483647;"
  >
    <div
      class="content"
      style="height: 0px; left: 0px; top: 0px; transform: scale(Infinity) translate(0px, 0px); transition-duration: 300ms; width: 0px;"
    >
      <img
        alt="foo"
        src="foo.jpg"
        width="500"
      />
    </div>
    <button
      aria-label="Unzoom image"
      class="trigger btn"
    />
  </div>
</body>
`;

exports[`unzooms on scroll 2`] = `
<body>
  <div>
    <div
      class="wrap"
    >
      <img
        alt="foo"
        src="foo.jpg"
        width="500"
      />
      <button
        aria-label="Zoom image"
        class="trigger btn"
      />
    </div>
  </div>
</body>
`;

exports[`unzooms using ESC key 1`] = `
<body>
  <div>
    <div
      class="wrapHidden"
    >
      <img
        alt="foo"
        src="foo.jpg"
        width="500"
      />
      <button
        aria-label="Zoom image"
        class="trigger btn"
      />
    </div>
  </div>
  <div
    aria-modal="true"
    class="overlay"
    role="dialog"
    style="background-color: rgba(255, 255, 255, 0.95); transition-duration: 300ms; z-index: 2147483647;"
  >
    <div
      class="content"
      style="height: 0px; left: 0px; top: 0px; transform: scale(Infinity) translate(0px, 0px); transition-duration: 300ms; width: 0px;"
    >
      <img
        alt="foo"
        src="foo.jpg"
        width="500"
      />
    </div>
    <button
      aria-label="Unzoom image"
      class="trigger btn"
    />
  </div>
</body>
`;

exports[`unzooms using ESC key 2`] = `
<body>
  <div>
    <div
      class="wrapHidden"
    >
      <img
        alt="foo"
        src="foo.jpg"
        width="500"
      />
      <button
        aria-label="Zoom image"
        class="trigger btn"
      />
    </div>
  </div>
  <div
    aria-modal="true"
    class="overlay"
    role="dialog"
    style="background-color: rgba(255, 255, 255, 0.95); transition-duration: 300ms; z-index: 2147483647;"
  >
    <div
      class="content"
      style="height: 0px; left: 0px; top: 0px; transform: scale(Infinity) translate(0px, 0px); transition-duration: 300ms; width: 0px;"
    >
      <img
        alt="foo"
        src="foo.jpg"
        width="500"
      />
    </div>
    <button
      aria-label="Unzoom image"
      class="trigger btn"
    />
  </div>
</body>
`;

exports[`unzooms using ESC key 3`] = `
<body>
  <div>
    <div
      class="wrap"
    >
      <img
        alt="foo"
        src="foo.jpg"
        width="500"
      />
      <button
        aria-label="Zoom image"
        class="trigger btn"
      />
    </div>
  </div>
</body>
`;

exports[`unzooms using ESC key 4`] = `
<body>
  <div>
    <div
      class="wrap"
    >
      <img
        alt="foo"
        src="foo.jpg"
        width="500"
      />
      <button
        aria-label="Zoom image"
        class="trigger btn"
      />
    </div>
  </div>
</body>
`;

exports[`zooms/unzooms with custom open/close text & <img /> 1`] = `
<body>
  <div>
    <div
      class="wrapHidden"
    >
      <img
        alt="foo"
        src="foo.jpg"
        width="500"
      />
      <button
        aria-label="Open me"
        class="trigger btn"
      />
    </div>
  </div>
  <div
    aria-modal="true"
    class="overlay"
    role="dialog"
    style="background-color: rgba(255, 255, 255, 0.95); transition-duration: 300ms; z-index: 2147483647;"
  >
    <div
      class="content"
      style="height: 0px; left: 0px; top: 0px; transform: scale(Infinity) translate(0px, 0px); transition-duration: 300ms; width: 0px;"
    >
      <img
        alt="foo"
        src="foo.jpg"
        width="500"
      />
    </div>
    <button
      aria-label="Close me"
      class="trigger btn"
    />
  </div>
</body>
`;

exports[`zooms/unzooms with custom open/close text & <img /> 2`] = `
<body>
  <div>
    <div
      class="wrap"
    >
      <img
        alt="foo"
        src="foo.jpg"
        width="500"
      />
      <button
        aria-label="Open me"
        class="trigger btn"
      />
    </div>
  </div>
</body>
`;

exports[`zooms/unzooms with defaults & <img /> 1`] = `
<body>
  <div>
    <div
      class="wrap"
    >
      <img
        alt="foo"
        src="foo.jpg"
        width="500"
      />
      <button
        aria-label="Zoom image"
        class="trigger btn"
      />
    </div>
  </div>
</body>
`;

exports[`zooms/unzooms with defaults & <img /> 2`] = `
<body>
  <div>
    <div
      class="wrapHidden"
    >
      <img
        alt="foo"
        src="foo.jpg"
        width="500"
      />
      <button
        aria-label="Zoom image"
        class="trigger btn"
      />
    </div>
  </div>
  <div
    aria-modal="true"
    class="overlay"
    role="dialog"
    style="background-color: rgba(255, 255, 255, 0.95); transition-duration: 300ms; z-index: 2147483647;"
  >
    <div
      class="content"
      style="height: 0px; left: 0px; top: 0px; transform: scale(Infinity) translate(0px, 0px); transition-duration: 300ms; width: 0px;"
    >
      <img
        alt="foo"
        src="foo.jpg"
        width="500"
      />
    </div>
    <button
      aria-label="Unzoom image"
      class="trigger btn"
    />
  </div>
</body>
`;

exports[`zooms/unzooms with defaults & <img /> 3`] = `
<body>
  <div>
    <div
      class="wrap"
    >
      <img
        alt="foo"
        src="foo.jpg"
        width="500"
      />
      <button
        aria-label="Zoom image"
        class="trigger btn"
      />
    </div>
  </div>
</body>
`;
